Drivers/IWDG_driver/Src/IndependentWatchdog.cpp

```diff
@@ -0,0 +1,45 @@
+#include "IndependentWatchdog.h";
+#include "main.h"
+
+/**
+ * @brief Borrow the configuration of an already existing watchdog and set it
+ * to the prescaler and Reload values.
+ *
+ */
+IWDG_t::IWDG_t(IWDG_HandleTypeDef *watchdog) {
+	IWDG_InitTypeDef initDef = watchdog->Init;
+	this->iwdg_prescaler = initDef.Prescaler;
+	this->iwdg_reload = initDef.Reload;
+}
+
+/**
+ * @brief Initalize the driver using custom IWDG_Prescaler and realod values.
+ *
+ */
+IWDG_t::IWDG_t(uint8_t iwdg_prescaler, uint16_t iwdg_reload) {
```

Comment:
Any reason the Init function and this constructor cant be the same? Like, just do the Init function's logic in this constructor rather than have two separate functions?

Reply:
I have modified the driver so that setting prescaler and reload is the same as re-initializing the watchdog.